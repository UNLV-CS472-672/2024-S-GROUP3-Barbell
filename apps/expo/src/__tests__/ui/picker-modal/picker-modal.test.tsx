import React from 'react'
import { Text } from 'react-native'

import { render } from '@testing-library/react-native'

import PickerModal from '~/components/ui/picker-modal/picker-modal'

describe('PickerModal', () => {
  it('renders correctly and matches snapshot', () => {
    const mockOnBackdropPress = jest.fn()

    const tree = render(
      <PickerModal title='Select an Option' isVisible={true} onBackdropPress={mockOnBackdropPress}>
        <Text>Child content goes here</Text>
      </PickerModal>,
    )

    expect(tree.toJSON()).toMatchSnapshot()
  })
})
