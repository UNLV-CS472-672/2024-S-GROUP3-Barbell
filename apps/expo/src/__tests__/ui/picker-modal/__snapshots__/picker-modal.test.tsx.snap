// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PickerModal renders correctly and matches snapshot 1`] = `
<Modal
  animationType="none"
  deviceHeight={null}
  deviceWidth={null}
  hardwareAccelerated={false}
  hideModalContentWhileAnimating={false}
  onBackdropPress={[MockFunction]}
  onModalHide={[Function]}
  onModalWillHide={[Function]}
  onModalWillShow={[Function]}
  onRequestClose={[Function]}
  panResponderThreshold={4}
  scrollHorizontal={false}
  scrollOffset={0}
  scrollOffsetMax={0}
  scrollTo={null}
  statusBarTranslucent={false}
  supportedOrientations={
    [
      "portrait",
      "landscape",
    ]
  }
  swipeThreshold={100}
  transparent={true}
  visible={true}
>
  <View
    accessibilityState={
      {
        "busy": undefined,
        "checked": undefined,
        "disabled": undefined,
        "expanded": undefined,
        "selected": undefined,
      }
    }
    accessible={true}
    collapsable={false}
    focusable={true}
    onClick={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      {
        "backgroundColor": "black",
        "bottom": 0,
        "height": 1334,
        "left": 0,
        "opacity": 0,
        "position": "absolute",
        "right": 0,
        "top": 0,
        "width": 750,
      }
    }
  />
  <View
    collapsable={false}
    deviceHeight={null}
    deviceWidth={null}
    hideModalContentWhileAnimating={false}
    onBackdropPress={[MockFunction]}
    onModalHide={[Function]}
    onModalWillHide={[Function]}
    onModalWillShow={[Function]}
    panResponderThreshold={4}
    pointerEvents="box-none"
    scrollHorizontal={false}
    scrollOffset={0}
    scrollOffsetMax={0}
    scrollTo={null}
    statusBarTranslucent={false}
    style={
      {
        "flex": 1,
        "justifyContent": "center",
        "margin": 37.5,
        "marginBottom": 20,
        "transform": [
          {
            "translateY": 1334,
          },
        ],
      }
    }
    supportedOrientations={
      [
        "portrait",
        "landscape",
      ]
    }
    swipeThreshold={100}
  >
    <View
      style={
        [
          {
            "bottom": 100,
            "position": "absolute",
          },
        ]
      }
    >
      <View
        style={
          [
            {
              "backgroundColor": "#232323",
              "borderRadius": 12,
              "paddingBottom": 4,
              "paddingTop": 8,
              "width": 675,
            },
          ]
        }
      >
        <Text
          className="p-6 text-center text-xl font-bold text-white"
        >
          Select an Option
        </Text>
        <View />
      </View>
    </View>
  </View>
</Modal>
`;
